### PR TITLE
flow: Add OracleDB exporter

### LINKED ISSUE #3166: Flow component for oracledb integration
Bring over the static mode oracledb integration as a Flow component.

### PR DESCRIPTION
**Pull request by JonathanWamsley**
#### PR Description
- Adds a oracledb exporter to the list of supported flow integrations.

#### Which issue(s) this PR fixes

Fixes #3166 
#### PR Checklist

- [x] CHANGELOG updated
- [x] Documentation added
- [x] Tests updated

### CODE REVIEW

File: docs/sources/flow/reference/components/prometheus.exporter.oracledb.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+Omitted fields take their default values.
+
+| Name                | Type     | Description                                                  | Default | Required |
+|---------------------|----------|--------------------------------------------------------------|---------|----------|
+| `connection_string` | `secret` | The connection string used to connect to an Oracle Database. |         | yes      |
+| `max_idle_conns`    | `int`    | Number of maximum idle connections in the connection pool.   | `0`     | no       |
+| `max_open_conns`    | `int`    | Number of maximum open connections in the connection pool.   | `10`    | no       |
+| `query_timeout`     | `int`    | The query timeout in seconds.                                | `5`     | no       |
+
+[The oracledb_exporter running documentation](https://github.com/iamseth/oracledb_exporter/tree/master#running) show the format and examples of the `connection_string` argument:
```
[the remark this answers is not in the document]
Comment (JonathanWamsley) [reply docs/sources/flow/reference/components/prometheus.exporter.oracledb.md]: updated.

File: component/prometheus/exporter/oracledb/oracledb.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+		Exports: exporter.Exports{},
+		Build:   exporter.New(createExporter, "oracledb"),
+	})
+}
+
+func createExporter(opts component.Options, args component.Arguments) (integrations.Integration, error) {
+	a := args.(Arguments)
+	return a.Convert().NewIntegration(opts.Logger)
+}
+
```
Comment (jcreixell) [inline component/prometheus/exporter/oracledb/oracledb.go:29]: This is something very new, but flow prometheus exporter components were missing the customized `instance` label required for compatibility with Grafana Cloud integrations. We have introduced them in #3873. Would you mind adding a function similar to https://github.com/grafana/agent/blob/f53714a7df98cb03185202f4caa839e69d8520dd/component/prometheus/exporter/mssql/mssql.go#L31-L42 and passing it like it is done [here](https://github.com/grafana/agent/blob/f53714a7df98cb03185202f4caa839e69d8520dd/component/prometheus/exporter/mssql/mssql.go#L22)?

The logic that the function needs to implement is defined [here](https://github.com/grafana/agent/blob/f53714a7df98cb03185202f4caa839e69d8520dd/pkg/integrations/oracledb_exporter/oracledb_exporter.go#L78-L84).


Here are some example [tests](https://github.com/grafana/agent/blob/f53714a7df98cb03185202f4caa839e69d8520dd/component/prometheus/exporter/mssql/mssql_test.go#L129-L151).

**Review by jcreixell — CHANGES_REQUESTED** [review]
please see my comment, and sorry for the last minute request

File: component/prometheus/exporter/oracledb/oracledb.go (hunk 2 above)
Comment (JonathanWamsley) [reply component/prometheus/exporter/oracledb/oracledb.go:30]: No problem. Updated to your suggestions @jcreixell !

**Review by spartan0x117 — APPROVED** [review]
LGTM!

**Review by jcreixell — APPROVED** [review]

**Pull request merged**